mapr/datadog_checks/mapr/mapr.py

```diff
@@ -0,0 +1,153 @@
+                break
+
+            if msg.error() is None:
+                # Metric received
+                try:
+                    metric = json.loads(ensure_unicode(msg.value()))[0]
+                    metric_name = metric['metric']
+                    if self.should_collect_metric(metric_name):
+                        # Will sometimes submit the same metric multiple time, but because it's only
+                        # gauges and monotonic_counter that's fine.
+                        self.submit_metric(metric)
+                        submitted_metrics_count += 1
+                except Exception as e:
+                    self.log.warning("Received unexpected message %s, wont be processed", msg.value())
+                    self.log.exception(e)
+            elif msg.error().code() != ck.KafkaError._PARTITION_EOF:
+                # Real error happened
+                raise CheckException(msg.error())
+
+        if not submitted_metrics_count and not self.is_first_check_run:
+            self.log.error("No metric to fetch in topic {}. Double-check the topic name and path", self.topic_path)
+
+        self.gauge(METRICS_SUBMITTED_METRIC_NAME, submitted_metrics_count, self.base_tags)
+        self.service_check(SERVICE_CHECK, AgentCheck.OK, self.base_tags + ['topic:{}'.format(self.topic_path)])
+        self.is_first_check_run = False
+
+    def get_connection(self):
+        if self._conn:
+            return self._conn
+
+        self._conn = ck.Consumer(
+            {
+                "group.id": "dd-agent",  # uniquely identify this consumer
+                "enable.auto.commit": False  # important, do not store the offset for this consumer,
+                # if we do it just once the mapr library has a bug (feature?) which prevents resetting it to the head
+            }
+        )
+        self._conn.subscribe([self.topic_path])
+        return self._conn
+
+    def should_collect_metric(self, metric_name):
+        if metric_name not in ALLOWED_METRICS:
+            # Metric is not part of datadog allowed list
+            return False
+        if not self.allowed_metrics:
+            # No filter specified, allow everything
+            return True
+
+        for reg in self.allowed_metrics:
+            if re.match(reg, metric_name):
+                # Metric matched one pattern
+                return True
+
+        self.log.debug("Ignoring non whitelisted metric: %s", metric_name)
+        return False
+
+    def submit_metric(self, metric):
+        metric_name = metric['metric']
+        tags = self.base_tags + ["{}:{}".format(k, v) for k, v in iteritems(metric['tags'])]
+
+        if 'buckets' in metric:
+            for bounds, value in metric['buckets'].items():
+                lower, upper = bounds.split(',')
+                self.submit_histogram_bucket(
+                    metric_name, value, int(lower), int(upper), monotonic=True, hostname=self.hostname, tags=tags
```

Comment:
Is it ok to use `monotonic=True` for all cases ?

Reply:
updated

Reply:
I mean, should that be something specific for each metric ?

Reply:
Sorry that was clear in my head but just in my head 🤦‍♂ 
What I wanted to say is that, there's only one bucket metric called `mapr.db.table.latency` which is always `non-monotonic`.
If new metrics that have a differnet monotoniticity are added in the future we will have to add custom logic to handle this. But right now there's no doc about it and thus nothing more we can do beside handle that single metric correctly